.github/workflows/test_client_with_new_server.yml

```diff
@@ -0,0 +1,68 @@
+name: Cross-compatibility with new Infinispan Server
+
+on:
+  schedule:
+    - cron: '0 2 * * *' # Runs every night at 2:00 AM UTC
+  workflow_dispatch:
+  push:
+
+jobs:
+  nightly-build:
+    runs-on: ubuntu-latest
+    env:
+      ISPN_SERVER_VER: 15.2.5.Final
```

Comment:
Maybe, instead of hardcoding the release, we could use 
`gh release view --json name|jq -r .name` to get the latest